dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataset/Section.java

```diff
@@ -109,7 +112,11 @@ public boolean hasCategoryCombo()
             }
         }
 
-        return categoryCombos;
+        List<CategoryCombo> sortedCategoryombos = categoryCombos.stream().collect( Collectors.toList() );
+
+        Collections.sort( sortedCategoryombos, new CategoryComboComparator() );
```

Comment:
I think you could simplify the code like this

```suggestion
        List<CategoryCombo> sortedCategoryombos = categoryCombos.stream().sorted(new CategoryComboComparator()).collect( Collectors.toList() );
```